syntax = "proto3";

message addMsg {
    int32 a = 1;
    int32 b = 2;
}

message rawImageMsg {
    bytes img=1;
}

message dotProductMsg {
    repeated float a = 1;
    repeated float b = 2;
}
  
message jsonImageMsg {
    string jsonimg=1;
}

message addReply {
    int32 sum = 1;
}

message dotProductReply {
    float dotproduct = 1;
}

message imageReply {
    int32 width=1;
    int32 height=2;
}

service RawImage {
    rpc RawImageDimens(rawImageMsg) returns (imageReply) {};
};

service Add {
    rpc AddNums(addMsg) returns (addReply) {};
};

service DotProduct {
    rpc DotProd(dotProductMsg) returns (dotProductReply) {};
};

service JsonImage {
    rpc JsonImageDimens(jsonImageMsg) returns (imageReply) {};
};